js/util/extractProps.ts

```diff
@@ -6,11 +8,16 @@ export interface Component {
   props: {
     [propertyKey: string]: any;
   };
+  heapOptions?: ClassHeapOptions;
+}
+
+export interface ClassHeapOptions {
+  eventProps?: PropExtractorCriteria;
 }
 
 export interface PropExtractorCriteria {
   include: string[];
-  exclude: string[];
+  exclude?: string[];
```

Comment:
If `exclude` is optional, shouldn't `include` also be optional?  E.g., a higher priority config might only be concerned with excluding props, while a lower priority config might only be concerned with including props.

The way the include/exclude arrays should be represented should be one the following for both arrays:
* `undefined` means no props in that array
* empty array means no props in that array

When `include` is required, but `exclude` is optional, the way "no props" can be represented isn't consistent.

Reply:
No, I don't think so - I put the comment in the RFC on paper, but I'll copy it here in case you see this first:
No.  I think that explicit exclusions will not be often used, and so I’d like to avoid having the programmer need to write it each time.  So that should be optional. However, in the vast majority of cases, it’s pointless to do this without an inclusion set, and it will nearly always be there, so that should be required.  It may be empty, but there’s little gained by allowing it to be omitted, and we force any code that touches it to have to check that it’s defined.

Reply:
I still disagree.  Although we expect that criteria with only `include`s will be more common than only `exclude`s, this won't always be the case.  If we include builtin props that the dev needs to exclude, the criteria will likely only need an `exclude`.  It's inconsistent to make one of these required/optional but not both.  To me, consistency with this sort of thing is important when judging the quality of an API.

Adding code that checks that `include` is defined is cheap, especially if it's already checking whether `exclude` is defined.

Reply:
Left as-is per discussion.